src/Compilers/CSharp/Portable/Binder/BindingDiagnosticBag.cs

```diff
@@ -0,0 +1,165 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal static class BindingDiagnosticBagExtensions
```

Comment:
>BindingDiagnosticBagExtensions [](start = 26, length = 30)

File is incorrectly named given this class name.

Reply:
> File is incorrectly named given this class name.

The BindingDiagnosticBag class is below

---
In reply to: [361546561](https://github.com/dotnet/roslyn/pull/40560#discussion_r361546561) [](ancestors = 361546561)

Reply:
Please place the class `BindingDiagnosticBagExtensions` in a file `BindingDiagnosticBagExtensions.cs`.

---
In reply to: [361549090](https://github.com/dotnet/roslyn/pull/40560#discussion_r361549090) [](ancestors = 361549090,361546561)

Reply:
Added a PROTOTYPE comment in [https://github.com/dotnet/roslyn/pull/40640](https://github.com/dotnet/roslyn/pull/40640)

---
In reply to: [361554376](https://github.com/dotnet/roslyn/pull/40560#discussion_r361554376) [](ancestors = 361554376,361549090,361546561)